src/pg/sql/11_kmeans.sql

```diff
@@ -0,0 +1,31 @@
+CREATE OR REPLACE FUNCTION  CDB_KMeans(query text, no_clusters integer,no_init integer default 20)
+RETURNS table (cartodb_id integer, cluster_no integer) as $$
+    
+    import plpy 
+    plpy.execute('SELECT cdb_crankshaft._cdb_crankshaft_activate_py()')
+    from crankshaft.clustering import kmeans
+    return kmeans(query,no_clusters,no_init)
+
+$$ language plpythonu;
+
+CREATE OR REPLACE FUNCTION CDB_WeightedMean(query text, weight_column text, category_column text default null )
+RETURNS table (the_geom geometry,class integer ) as $$
```

Comment:
Why do we limit this to be an integer?

Example for this: I would like to have the neighbourhood name as the category. Does it make sense?

Wouldn't be better to cast to `::text` so we can support almost any column type? 

I don't know a better way because using polymorphic types would force us to have a more complex function signature, which I don't want.